### PR TITLE
feat(w3-patients): create patient endpoint, validator

### PR DESCRIPTION
**Pull request by 14mdzk**
1. Create patient endpoint
2. Phone number validator
3. Date validator

What needs to be done / next task:
- adding url validator for identityCardScanImg field
- browse patient endpoint

## Summary by CodeRabbit

- **New Features**
  - Introduced a controller for managing medical record patient operations.
  - Implemented new API routes for medical record patient management.

- **Improvements**
  - Updated data structures and interfaces for better clarity and functionality in medical record patient management.

- **Bug Fixes**
  - Ensured accurate validation of phone numbers based on country codes and formats.

### CODE REVIEW

File: w3/protocol/api/route/public_route.go [hunk 1]
```diff
@@ -25,20 +25,27 @@ type PublicRouteParams struct {
...
 
 	// Create routes group.
 	route := params.App.Group("/v1")
 	route.Get("/ping", pingController.GetPingController)
 	route.Post("/user/it/register", userITController.RegisterUserIT)
 	route.Post("/user/it/login", userITController.LoginUserIT)
 
+	medical := route.Group("/medical")
+	medical.Post("/patient", medicalRecordPatientController.Create)
+
```
Comment (enjaytarigan) [inline w3/protocol/api/route/public_route.go:48]: Medical record feature should in private routes

File: w3/protocol/api/controller/medical_record_patient_controller.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	return &medicalRecordPatientController{patientUsecase}
+}
+
+func (ctr *medicalRecordPatientController) Create(c *fiber.Ctx) error {
+	var request usecase.MedicalRecordPatientCreateRequest
+
+	if err := c.BodyParser(&request); err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"error": "Cannot parse JSON",
+		})
```
Comment (enjaytarigan) [inline w3/protocol/api/controller/medical_record_patient_controller.go:34]: 1. We already global error handler, just return fiber.StatusBadRequest (see: user_it_controller.go for usage example)

File: w3/protocol/api/controller/medical_record_patient_controller.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	if err := helper.ValidateStruct(&request); err != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"error": err.Error(),
+		})
```
Comment (enjaytarigan) [inline w3/protocol/api/controller/medical_record_patient_controller.go:40]: We already global error handler, it also handle error from go-validator

File: w3/protocol/api/controller/medical_record_patient_controller.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	data, err := ctr.patientUsecase.Create(request)
+	if err != nil {
+		return c.Status(http.StatusInternalServerError).JSON(err)
+	}
```
Comment (enjaytarigan) [inline w3/protocol/api/controller/medical_record_patient_controller.go:46]: The global error handler also handles error from usecase and domain layer

**Review by enjaytarigan — CHANGES_REQUESTED** [review]

File: w3/protocol/api/controller/medical_record_patient_controller.go (hunk 2 above)
Comment (14mdzk) [reply w3/protocol/api/controller/medical_record_patient_controller.go:34]: done in next commit, forgot to merge the latest one.

File: w3/protocol/api/controller/medical_record_patient_controller.go (hunk 3 above)
Comment (14mdzk) [reply w3/protocol/api/controller/medical_record_patient_controller.go:40]: done in next commit, forgot to merge the latest one.

File: w3/protocol/api/controller/medical_record_patient_controller.go (hunk 4 above)
Comment (14mdzk) [reply w3/protocol/api/controller/medical_record_patient_controller.go:46]: done in next commit, forgot to merge the latest one.

**Pull request merged**